cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/BaseOAuthWrapperController.java

```diff
@@ -106,24 +107,28 @@ protected OAuthWebApplicationService createService(final RegisteredService regis
      * Create an authentication from a user profile.
      *
      * @param profile the given user profile
+     * @param service the registered service
      * @return the built authentication
      */
-    protected Authentication createAuthentication(final UserProfile profile) {
```

Comment:
Following up, that I do think this functionality needs to be refactored for CAS 5, but perhaps after the boot changes go in. We could get started without boot, and I likely can handle the merge later.